aggregator.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import Queue
+
+from Core.message_object import BeliefMessage
+from Core.process import Process
+from Core.utils import PhasedLoopLimitter
+
+from belief_manager import BeliefManager
+
+
+logger = logging.getLogger(__name__)
+
+
+class Aggregator(Process):
+  """ Aggregates data from both the radio and the autopilot, and chooses the
+  best to send along to the BeliefManager. """
+
+  def __init__(self, mav_queue, radio_queue, belief_box):
+    """ Args:
+      mav_queue: The queue for receiving autopilot data.
+      radio_queue: The queue for receiving radio data.
+      belief_box: The mailbox for sending data to the BeliefManager process. """
+    self.__limitter = PhasedLoopLimitter(BeliefManager.CYCLE_LENGTH)
+
+    self.__mav_queue = mav_queue
+    self.__radio_queue = radio_queue
+    self.__belief_box = belief_box
+
+    self.__clear()
+
+  def __clear(self):
+    """ Clears all the accumulators and buffers. """
+    # Buffer for storing radio data.
+    self.__radio_buffer = []
+
+    # Accumulators for various positional measurements.
+    self.__latitude = 0.0
+    self.__longitude = 0.0
+    self.__x_vel = 0.0
+    self.__y_vel = 0.0
+    self.__message_count = 0
+
+  def iterate(self):
+    """ Runs a single iteration of the process. This should be synched with the
+    frequency at which BeliefManager is run. """
+    logger.debug("Running iteration.")
+    # Clear everything.
+    self.__clear()
+
+    # Get and buffer radio data.
+    while True:
+      try:
+        radio_message = self.__radio_queue.get_nowait()
+      except Queue.Empty:
```

Comment:
I think I used it because I was worried about other methods not being reliable. Now that I think about it, though, non-reliability is not a huge problem here, and really shouldn't be much of an issue at all with our SPSC use-case.